joomla/joomla-cms
# Correct access checks for Categories Quick Icon

Pull Request for Issue # .

### Summary of Changes
For the "Article Categories" quick icon on the dashboard the incorrect access checks are made. Right now the permission is checked against `com_categories`, which is not a valid asset in Joomla. As the icon links to the Article categories the correct permission check should be made against `com_content`. This PR implements this.

### Testing Instructions
1. Create a new user group with "Public" as the parent user group. Allow actions "Administrator Login", "Access Administration Interface" and "Create" in the Global configuration. 
2. Login with as test user assigned to this new user group only and confirm that "Articles" and "Article Categories" are visible in the control panel. 
<img width="687" alt="Screenshot 2021-09-17 at 16 06 25" src="https://user-images.githubusercontent.com/522834/133798427-957e6202-236d-46f1-832d-d3f996576485.png">

3. Now go to the component options of the article manager and set "Access Administration Interface" and "Create" to denied for the test user group. 
4. Go back to the control panel with the test user and confirm that "Article Categories" is still visible, clicking on the icon will give an `403 You don't have permission to access this.` error.
<img width="686" alt="Screenshot 2021-09-17 at 16 09 33" src="https://user-images.githubusercontent.com/522834/133798460-217841c9-d0c0-46b8-a69b-09f84442a44a.png">

5. Apply patch.
6. Go back to the control panel with the test user and confirm that "Article Categories" is no longer visible.
<img width="672" alt="Screenshot 2021-09-17 at 16 22 04" src="https://user-images.githubusercontent.com/522834/133798573-d04d2250-ce07-4b47-b510-e916ebf21e7c.png">

7. Go to the component options of the article manager and set "Access Administration Interface" and "Create" to inherited (so allowed) for the test user group.  
8. Go back to the control panel with the test user and confirm that "Articles" and "Article Categories" are both visible in the control panel.
